### PR TITLE
Export Dev Submissions to CSV

### PR DESCRIPTION
**Pull request by airaagrawal1**
### Summary 

This pull request implemented "export to csv" functionality for dev leads to easily grade dev portfolio submissions. It includes the columns `name`, `netid`, `opened_score`, `reviewed_score`, and `total_score`. Submissions are sorted by netid and the csv only includes one submission from each member. 

If a dev portfolio assignment has no submissions, an error message will pop-up. 

Screen Shot 2022-10-13 at 1 01 53 PM

### Test Plan 

- Manually tested and checked that all submissions were present in CSV with correct values.
- All submissions were correctly sorted by netID.
- Tested dev portfolio with multiple submissions from one member. 
- Tested dev portfolio with no submissions.

### CODE REVIEW

File: frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx [hunk 1]
```diff
@@ -8,13 +10,64 @@ type Props = {
...
+  const handleExportToCsv = () => {
+    if (portfolio?.submissions === undefined || portfolio?.submissions.length <= 0) {
+      Emitters.generalError.emit({
+        headerMsg: 'Failed to export Dev Portfolio Submissions to CSV',
+        contentMsg: 'Please make sure there is at least 1 submission in the table.'
+      });
+      return;
+    }
+
+    const uniqueIds: string[] = [];
```
Comment (henry-li-06) [inline frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx:32]: This should probably be a set

File: frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx [hunk 2]
```diff
@@ -8,13 +10,64 @@ type Props = {
...
+    const sortedSubmissions = sortSubmissions(uniqueSubmissions);
+
+    const csvData = sortedSubmissions.map((submission) => ({
+      name: `${submission.member.firstName} ${submission.member.lastName}`,
+      netid: submission.member.netid,
+      opened_score: Number(submission.openedPRs.some((pr) => pr.status === 'valid')),
+      reviewed_score: Number(submission.reviewedPRs.some((pr) => pr.status === 'valid')),
+      total_score:
+        Number(submission.reviewedPRs.some((pr) => pr.status === 'valid')) +
+        Number(submission.openedPRs.some((pr) => pr.status === 'valid'))
```
Comment (henry-li-06) [inline frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx:52]: The opened score and reviewed score only needs to be computed to once and can be stored in a var probably

File: frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx [hunk 3]
```diff
@@ -29,6 +82,7 @@ const DevPortfolioDetails: React.FC<Props> = ({ uuid, isAdminView }) => {
       <Header textAlign="center" as="h3">
         Deadline: {new Date(portfolio.deadline).toDateString()}
       </Header>
+      <Button onClick={() => handleExportToCsv()}>Export to CSV</Button>
```
Comment (JacksonStaniec) [inline frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx:85]: We should only show this button if `isAdminView`.

**Comment by henry-li-06** [comment]
Is it possible to change the name of the downloaded csv?

**Comment by airaagrawal1** [comment]
> Is it possible to change the name of the downloaded csv?

Yes there is! Is '<name of dev portfolio> submissions' good?

**Comment by henry-li-06** [comment]
> > Is it possible to change the name of the downloaded csv?
> 
> Yes there is! Is ' submissions' good?

Maybe something like `<dev-portfolio-name>-submissions`?

**Comment by airaagrawal1** [comment]
> > > Is it possible to change the name of the downloaded csv?
> > 
> > 
> > Yes there is! Is ' submissions' good?
> 
> Maybe something like `<dev-portfolio-name>-submissions`?

lmao thats what I had typed out but idk why it didnt show up. Will do!

**Review by henry-li-06 — APPROVED** [review]
LGTM

**Pull request merged**